pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -230,20 +237,40 @@ func (a *Operator) providedAPIsFromCSVs(group *v1.OperatorGroup, logger *logrus.
 			logger.WithError(err).Warn("could not create OperatorSurface from csv")
 			continue
 		}
-		providedAPIsFromCSVs = providedAPIsFromCSVs.Union(operatorSurface.ProvidedAPIs().StripPlural())
+		for providedAPI := range operatorSurface.ProvidedAPIs().StripPlural() {
+			providedAPIsFromCSVs[providedAPI] = csv
```

Comment:
Why lug the CSV around when you can filter out the ones you don't want [here](https://github.com/operator-framework/operator-lifecycle-manager/blob/1debd997de2f70ccfc7f4e9c82726aba7c36aa60/pkg/controller/operators/olm/operatorgroup.go#L225)?

Reply:
I don't think I want to filter out any CSVs. The CSVs are returned so that in pruneProvidedAPIs while iterating over the provided apis, the CSV phase for a given api can be checked to ensure it has progressed to the point of validation (past phase pending).

Reply:
hmm, I think they are roughly isomorphic, but this isn't a big deal.